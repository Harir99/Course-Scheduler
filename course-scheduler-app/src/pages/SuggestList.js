import { React } from 'react'
import { Button, Card, ListGroup } from 'react-bootstrap'
import '../index.css'

function SuggestList (props) {
  if (props.chosen < 5) {
    return (
            <Card className={`text-left ${(props.suggestions.length !== 0) ? 'scrollable' : ''}`}>
                <ListGroup variant="flush">
                    {props.suggestions.map((item) => (
                        <ListGroup.Item key={item}>
                                <Card.Body>
                                    <Card.Text>
                                        {item}
                                    </Card.Text>
                                        <Button value={item}
                                                onClick={props.addCourse}
                                                size="sm"
                                                style={{
                                                  backgroundColor: '#84428C',
                                                  border: '#84428C'
                                                }}>
                                            Add Course
                                        </Button>
                                </Card.Body>
                        </ListGroup.Item>
                    ))}
                </ListGroup>
            </Card>
    )
  }
}

export default SuggestList
