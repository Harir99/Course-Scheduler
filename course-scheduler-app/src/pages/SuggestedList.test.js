import SuggestList from './SuggestList'
import renderer from 'react-test-renderer'
import React from 'react'

test('tests that the suggested list can be rendered', () => {
  const tree = renderer.create(<SuggestList suggestions={['ACCT*1220*0101 (6573) Intro Financial Accounting']} chosen={0} addCourse={[]}/>).toJSON()
  expect(tree).toMatchSnapshot()
})
