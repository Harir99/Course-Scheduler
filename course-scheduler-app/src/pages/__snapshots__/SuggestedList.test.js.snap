// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests that the suggested list can be rendered 1`] = `
<div
  className="text-left scrollable card"
>
  <div
    className="list-group list-group-flush"
    onKeyDown={[Function]}
  >
    <div
      className="list-group-item"
      data-rr-ui-event-key={null}
      onClick={[Function]}
    >
      <div
        className="card-body"
      >
        <p
          className="card-text"
        >
          ACCT*1220*0101 (6573) Intro Financial Accounting
        </p>
        <button
          className="btn btn-primary btn-sm"
          disabled={false}
          onClick={Array []}
          style={
            Object {
              "backgroundColor": "#84428C",
              "border": "#84428C",
            }
          }
          type="button"
          value="ACCT*1220*0101 (6573) Intro Financial Accounting"
        >
          Add Course
        </button>
      </div>
    </div>
  </div>
</div>
`;
